bin/yaml/triton.yaml

```diff
@@ -0,0 +1,22 @@
+compilers:
+  triton:
+    type: tarballs
+    dir: triton-{name}
+    depends:
+      - tools/cmake 3.29.2
+      - tools/ninja 1.11.1
+      - tools/wheel 0.43.0
+      - tools/pybind11 2.13.5
+      - compilers/c++/clang 18.1.0
+      - libraries/triton/triton-shared main
+    compression: gz
+    check_exe: python3 venv/lib/python3.12/site-packages/triton/tools/compile.py -h
+    url: https://github.com/triton-lang/triton/archive/refs/tags/v{name}.tar.gz
+    targets:
+      - 2.1.0
+    after_stage_script:
+      - cd triton-{name}
+      - python3 -m venv venv
+      - source venv/bin/activate
+      - cd python
+      - TRITON_BUILD_WITH_CLANG_LLD=true TRITON_PLUGIN_DIRS=/opt/compiler-explorer/triton-shared-main pip install -e .
```

Comment:
This will build on our "admin" installer node, which isn't a great place for this to go. In other languages we _try_ and build a "hermetic", relocatable tarball that we use as the source of truth.

From (bitter) experience in my day job, python and its various virtual environments aren't great at this kind of thing. `pip` and `poetry` don't handle native dependencies at all, `conda` is not relocatable.

Also hardcoding `/opt/compiler-explorer` here is not a great idea as we support other installation locations (though pragmatically this will work for the CE main install)